[Epic#3] 로그인 된 사용자는 마이페이지로 이동 할 수 있다.

happydhKim opened the pull request:
# [Epic#3] 로그인 된 사용자는 마이페이지로 이동 할 수 있다. 

## 해당 이슈 📎

- [Epic#3] 로그인 된 사용자는 마이페이지로 이동 할 수 있다.  (#41)

## 변경 사항 🛠

- 마이페이지 ui 구현
- server-web에 스키마 구현

## 테스트 ✨

> 없음
## 리뷰어 참고 사항 🙋‍♀️

- 전반적으로 UI만 작업이 이루어 진 상황입니다.

happydhKim committed:
[ FEATURE ] 히스토리에 대한 스키마 정의 (#41)

Why
- 사용자의 히스토리를 담아 둘 테이블이 필요하다.

What
- 히스토리에 대한 스키마를 정의 및 구현.

happydhKim committed:
[ FEATURE ] 마이페이지 메인 UI 구현 (#41)

Why
- 사용자의 마이페이지가 필요하다.
- 마이페이지는 라우터에 연동 되어야 한다.

What
- 마이페이지 기본 UI 구현.
- 마이페이지는 사용자 정보를 담는 UserInfo와 히스토리가 담겨있는 UserHistory로 나눠서 구현.
- 라우터에 마이페이지 연동.

happydhKim committed:
[ MERGE ] develop 브랜치 작업 feature/issue41로 가져옴

* develop -> feature/issue41

happydhKim committed:
[ FEATURE ] 마이페이지 내에 유저 정보 UI 세부 구현 (#41)

Why
- 유저 정보 내에 프로필 및 유저 정보를 나타내는 UI가 필요하다.
- 유저 정보 내에 히스토리를 선택 할 수 있는 버튼이 필요하다.

What
- 유저 정보 내에 유저 정보를 나타내는 UI 구현.
- 유저 정보 내에 히스토리를 선택 할 수 있는 버튼 구현.

happydhKim committed:
[ REFACTOR ] 마이페이지에 회원 정보 UI 리펙토링

Why
- 회원 정보가 기존 페이지에 어울리지 않아 수정해야함.

What
- 기존 페이지에 어울리게 버튼 색상과 위치를 변경.
- 회원 정보 UI를 개선함.
- setting 버튼을 추가함.

happydhKim committed:
[ FEATURE ] 마이페이지에서 사용자 정보 세팅 팝업 UI 구현 (#41)

Why
- 사용자의 정보를 수정할 수 있는 페이지가 필요하다.

What
- 사용자 정보를 수정할 수 있는 팝업 UI 구현.
- 머티리얼 UI를 이용해 구현.

happydhKim committed:
[ FEATURE ] 히스토리 카드 UI 구성 (#41)

Why
- 마이페이지에서 히스토리를 보여줄 리스트가 필요하다.

What
- 히스토리 카드 UI 기본 세팅 구현.

happydhKim committed:
[ FEATURE ] 히스토리 유저와 채널을 고려한 스키마 작성 (#41)

Why
- 히스토리를 조회하기 위한 스키마가 필요하다.

What
- 히스토리 qraphql 스키마 구현.
- mongodb 스키마 수정. ( 변화하는 값을 삭제 )

happydhKim committed:
[ REFACTOR ] 팀원 컨벤션 수정 (#41)

Why
- 스타일 관련 컨벤션이 바뀌어 수정해야 함 (Wrapper 관련 )

What
- 관련된 스타일을 수정함.

aereeeee commented on web/src/components/history/UserHistory/style.jsx:
```diff
@@ -0,0 +1,17 @@
+    flex: 1;
+    height: 100%;
+    overflow: auto;
+    border-left: 1px solid ${(({ theme }) => theme.palette.dropyGray[3])};
```
colorGray theme유틸을 사용해 주시면 좋을것 같습니다~!

happydhKim replied:
넵🤠

aereeeee commented on web/src/components/history/UserHistoryCard/style.jsx:
```diff
@@ -0,0 +1,39 @@
+    width: 80px;
+    height: 80px;
+    border-radius: 100%;
+    border: 1px solid ${(({ theme }) => theme.palette.dropyGray[3])}; 
```
colorGray와 px을 적극적으로 사용해주세용~~

happydhKim replied:
감사해요 수정해 놓겠습니다! 적극 활용해보겠습니다👩🏻‍🎨

devjaewon commented on server-api/src/models/histories.js:
```diff
@@ -0,0 +1,45 @@
+});
+
+HistorySchema.statics.upsert = function upsertHistory(userId, channelId) {
+  const HistoryModel = this;
```
`upser` 메서드 만든 방식 매우 좋은것 같습니다! 👍

happydhKim replied:
감사합니다!

aereeeee reviewed (commented):
시간이 넉넉하지 않아 간단히 리뷰 남깁니다!
작업을 많이 하셨네요! 짱입니다 수고하셨습니다~~

happydhKim committed:
[ REFACTOR ] 리뷰어에게 받은 리뷰사항 적용

Why
- 히스토리 스키마에서 찾은 에러를 수정해야 함.

What
- 히스토리 스키마에서 히스토리를 수정하는 부분을 수정.

happydhKim committed:
[ REFACTOR ] 리뷰어에게 전달받은 스타일 관련 수정 (#41)

Why
- 스타일 관련 px와 전역 변수를 사용 안한 곳에 대해 수정이 필요하다.

What
- px는 모두 함수로 수정함.
- 색은 전역변수로 모두 수정함.
